feat(knowledge): support doc2x, mistral, MacOS, MinerU... OCR

eeee0717 opened the pull request:
- [x] OCR翻译pdf文件


https://github.com/user-attachments/assets/357e9f17-360b-4c66-9039-37af4ef851e4

eeee0717 committed:
chore

eeee0717 committed:
feat: knowledge ocr settings

eeee0717 committed:
feat: support doc2x ocr

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

DeJeune commented:
本来我在写mistral的，等你PR提交了我在写后面的吧

eeee0717 commented:
> 本来我在写mistral的，等你PR提交了我在写后面的吧

OK，今天刷讨论的时候看到你发的了，本来想问你一下，忘记了🤦

eeee0717 committed:
feat(knowledge-ocr): optimize and clean code

eeee0717 committed:
chore

DeJeune committed:
Merge branch 'main' into local-pr-3734

DeJeune committed:
refactor(Settings): Refactored knowledge settings to utilize the new OCR provider management.

DeJeune committed:
refactor(fileService): integrate Mistral and Gemini file services

- Added Mistral and Gemini file services to handle file uploads, retrieval, listing, and deletion.
- Refactored IPC handlers to support new file service architecture.
- Updated preload API to accommodate new file service methods.
- Removed deprecated Gemini service implementation.
- Enhanced type definitions for file handling to support local and remote sources.

DeJeune committed:
feat(OCR): implement Mistral OCR provider and enhance file handling

- Introduced MistralOcrProvider for processing OCR tasks with Mistral API.
- Updated BaseOcrProvider to return LocalFileSource in parseFile method.
- Enhanced file upload and retrieval logic in MistralService.
- Added support for local file source handling in various services.
- Updated preload API and type definitions to accommodate new OCR provider functionalities.

DeJeune committed:
refactor(icons): update icon components

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

DeJeune committed:
feat(OCR): add MacSysOcrProvider for macOS OCR functionality

- Introduced MacSysOcrProvider to handle OCR tasks using the mac-system-ocr library.
- Enhanced the OcrProviderFactory to support the new macOS OCR provider.
- Updated file handling to include base64 file processing.
- Added a new SVG icon for the macOS OCR provider.
- Updated the state management to include the new system OCR provider option.
- Improved user interface to conditionally display the system OCR provider based on the operating system.

DeJeune committed:
fix(FileStorage): update base64File method to return base64 string instead of Buffer

DeJeune committed:
merge 'main' into local-pr-3734

eeee0717 committed:
fix(FileStorage): add check for existing directory before deletion

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
fix: redux migrate error

eeee0717 committed:
chore

eeee0717 committed:
feat(OcrProvider): 统一文件格式

eeee0717 committed:
feat(KnowledgeService): 提取OCR预处理逻辑为单独方法

DeJeune committed:
Merge branch 'main' into local-pr-3734

DeJeune committed:
feat(OCR): improve PDF handling and update package versions

DeJeune committed:
chore(dependency): update mac-system-ocr and refine electron-builder configuration

DeJeune committed:
refactor(KnowledgeService): improve file name handling and remove debug logs from KnowledgeContent

DeJeune committed:
refactor(Knowledge): enhance progress tracking in UI components

DeJeune committed:
Merge branch 'main' into local-pr-3734

DeJeune committed:
feat(OCR): add recognition level and minimum confidence options for Mac system OCR provider

DeJeune committed:
chore(package): move @cherrystudio/mac-system-ocr to optionalDependencies

DeJeune committed:
chore(OCR): remove redundant logging statements

DeJeune committed:
feat(UI/files): integrate Mistral file handling and update OCR provider to use optional dependency

DeJeune committed:
feat(OCR): add options for recognition level and minimum confidence

kangfenmao commented:
等我测试，这个改动非常强👍

DeJeune committed:
Merge branch 'main' into feat-knowlege-ocr

DeJeune committed:
feat: enhance migration logic and add file type definitions

- Refactored migration logic in migrate.ts to improve error handling and state management.
- Introduced new file type definitions in file.ts, including interfaces for local and remote files, and type guards for file validation.
- Updated index.ts to export new file types and removed redundant file type definitions.
- Cleaned up file utility functions by removing unused base64 conversion function.
- Updated yarn.lock to include new dependencies and version updates for @mistralai/mistralai and @cherrystudio/mac-system-ocr.

DeJeune committed:
fix: update web search localization keys for consistency and clarity

DeJeune committed:
fix: update IPC handler for base64 file and enhance upload response structure in GeminiService

DeJeune committed:
Merge branch 'main' into feat-knowlege-ocr

DeJeune committed:
fix: update mac-system-ocr dependency to version 0.2.2

DeJeune committed:
fix: add macOS availability checks in MacSysOcrProvider and OcrProviderFactory

DeJeune committed:
fix: ensure MacOCR is correctly assigned after dynamic import in MacSysOcrProvider

DeJeune committed:
fix: add File_DeleteDir channel and update deleteDir invocation in preload API

DeJeune committed:
fix: add mocks for minapps configuration and window.api in export tests

DeJeune committed:
Merge branch 'main' into feat/ocr

DeJeune committed:
refactor: remove binaryImage method from api in preload index

DeJeune committed:
chore: update yarn.lock and modify useSettings hook

- Added new dependencies for mac-system-ocr and mistralai packages in yarn.lock.
- Removed unused topN setting from KnowledgeSettings component.
- Refactored useSettings hook to include setPinTopicsToTop method.

DeJeune committed:
fix(i18n): update localization for tools and web search settings

- Added new OCR settings under tools in English, Chinese (Simplified and Traditional) localization files.
- Updated web search settings to reflect new structure in localization files.
- Adjusted navigation path for web search settings in the Inputbar component.

DeJeune committed:
fix(i18n): update Japanese, Russian, Chinese, and Traditional Chinese localization files

DeJeune committed:
refactor: update file handling and dependencies

- Added '@cherrystudio/mac-system-ocr' to external dependencies in electron.vite.config.ts.
- Updated debug script in package.json to use 'electron-vite dev' for better debugging.
- Renamed 'File_Create' to 'File_CreateTempFile' in IpcChannel for clarity.
- Replaced 'adm-zip' with 'node-stream-zip' for zip file extraction in install-bun.js and install-uv.js scripts.
- Refactored file handling across various components to use 'FileMetadata' instead of 'FileType' for improved type consistency.
- Removed unused GeminiService and updated related file handling logic.
- Enhanced error handling and logging in file services and message services.

eeee0717 committed:
feat: add MinerU

eeee0717 committed:
chore: add i18n

eeee0717 committed:
chore: add rc version

DeJeune committed:
Merge branch 'feat-knowlege-ocr' of https://github.com/eeee0717/cherry-studio into feat/ocr

DeJeune committed:
refactor: streamline OCR processing in MacSysOcrProvider and update type definitions in MineruOcrProvider

- Removed batch processing in MacSysOcrProvider for direct page handling, improving efficiency.
- Updated type definitions from FileType to FileMetadata in MineruOcrProvider for consistency across the codebase.

DeJeune committed:
refactor: add OCR provider selection to KnowledgeSettings component

- Introduced a new settings item for selecting OCR providers in the KnowledgeSettings component.
- Removed the previous implementation of the OCR provider selection to streamline the UI.
- Updated the state management for the selected OCR provider to enhance user experience.

DeJeune committed:
Merge branch 'main' into feat/ocr

DeJeune committed:
chore: clean up yarn.lock by removing unused package entries

- Removed obsolete entries for 'caseless' and 'path-type' packages from yarn.lock to streamline dependencies and reduce clutter.

DeJeune committed:
refactor: update type definitions for file handling in FileStorage and FilesPage components

- Changed return type from FileType to FileMetadata in saveBase64Image method of FileStorage class for consistency.
- Updated import in FilesPage component to reflect the new FileMetadata type, enhancing clarity in file handling.

DeJeune committed:
refactor: remove unused imports in GeminiProvider for cleaner code

- Eliminated unnecessary imports such as FileState, Pager, and CacheService from GeminiProvider.ts to enhance code clarity and maintainability.

DeJeune committed:
refactor: remove unused file components and clean up imports

- Deleted GeminiFiles, MistralFiles, and RemoteFileList components to streamline the codebase.
- Cleaned up imports in AppUpdater.ts for improved organization and clarity.
- Updated Message.tsx to remove redundant import, enhancing code maintainability.

eeee0717 committed:
fix: fix doc2x extract zip error

eeee0717 committed:
rename ocr to preprocess

eeee0717 committed:
feat: change add knowledge ui

DeJeune committed:
refactor: simplify ToolSettings component layout

- Removed the SettingContainer wrapper for improved readability.
- Cleaned up imports and streamlined the rendering logic for WebSearchSettings and PreprocessSettings.

eeee0717 committed:
chore: remove unnecessary page

eeee0717 committed:
fix: preprocess cannot remove

eeee0717 committed:
feat: remove upload file

eeee0717 committed:
chore

eeee0717 committed:
chore: add i18n

eeee0717 committed:
chore

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
feat: split ocr and preprocess

eeee0717 committed:
chore: add i18n

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
chore: i18n

eeee0717 committed:
feat: improve ui and notification

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
feat: hide built in key

eeee0717 committed:
feat: add preprocess icon

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
chore

eeee0717 committed:
chore: set version 1.5.0-rc.1

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
fix

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
feat: add quota

eeee0717 committed:
feat: add quota

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
fix

eeee0717 committed:
fix: quota tag

eeee0717 committed:
feat: add quota tag

DeJeune committed:
chore: update package.json to include pdf-to-img dependency and add TODO comment in MacSysOcrProvider for future optimization

DeJeune committed:
Merge branch 'feat-knowlege-ocr' of https://github.com/eeee0717/cherry-studio into feat/ocr

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
fix: FileMetaData replace FileType

DeJeune committed:
feat: enhance development environment setup by ensuring userData path is set only once during initial setup

DeJeune committed:
Merge branch 'feat-knowlege-ocr' of https://github.com/eeee0717/cherry-studio into feat/ocr

eeee0717 committed:
fix: dev bug

eeee0717 committed:
Merge remote-tracking branch 'upstream/HEAD' into feat-knowlege-ocr

kangfenmao committed:
Merge branch 'main' into feat-knowlege-ocr

kangfenmao committed:
chore: version 1.5.0-rc.0

eeee0717 committed:
fix

DeJeune committed:
refactor: simplify development environment setup by removing redundant first setup check for userData path

DeJeune committed:
chore(package.json): reorder and reintroduce embedjs dependencies for consistency

DeJeune committed:
refactor(FilesPage): remove unused imports and simplify provider handling

- Eliminated unused imports including FilePdfOutlined and useProviders hook.
- Removed mapping of mistralProviders from the file type options, streamlining the component's logic.

eeee0717 committed:
Merge remote-tracking branch 'origin/feat-knowlege-ocr' into feat-knowlege-ocr

kangfenmao committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
fix: pdf-dist

eeee0717 committed:
fix: pdfjs-dist only use in macos

eeee0717 committed:
fix: add mineru key

eeee0717 committed:
fix

eeee0717 committed:
fix; move pdfjs to dependencies

eeee0717 committed:
fix: add canvas dependencies in linux

eeee0717 committed:
fix: add windows dependencies

eeee0717 committed:
fix: gtk url

eeee0717 committed:
Merge remote-tracking branch 'upstream/main' into feat-knowlege-ocr

eeee0717 committed:
fix: skip canvas rebuild

eeee0717 committed:
typo: excludeReBuildModules

DeJeune committed:
Merge remote-tracking branch 'origin/main' into feat/ocr

eeee0717 committed:
fix: release.yml

DeJeune committed:
chore: exclude 'pdfjs-dist' and 'canvas' from rebuild modules in electron-builder configuration

eeee0717 committed:
Merge remote-tracking branch 'origin/feat-knowlege-ocr' into feat-knowlege-ocr

eeee0717 committed:
fix

eeee0717 committed:
chore: change VITE_MINERU_API_KEY to RENDERER_VITE_MINERU_API_KEY

eeee0717 committed:
fix: use MAIN_VITE_MINERU_API_KEY and ocr only use in macos

eeee0717 committed:
fix: use MAIN_VITE_MINERU_API_KEY

DeJeune committed:
chore: remove macOS-only OCR package from build configuration and scripts

DeJeune committed:
Merge branch 'feat-knowlege-ocr' of https://github.com/eeee0717/cherry-studio into feat/ocr

eeee0717 committed:
fix: knowledge citation display

eeee0717 committed:
fix: move pdfjs-dist to dependencies

eeee0717 committed:
fix: add windows dependencies

DeJeune committed:
Merge branch 'main' into feat/ocr

DeJeune committed:
chore: remove duplicate pdfjs-dist entry from dependencies

eeee0717 committed:
Merge remote-tracking branch 'origin/feat-knowlege-ocr' into feat-knowlege-ocr

eeee0717 committed:
fix: remove windows dependencies

DeJeune committed:
chore: add excludeReBuildModules for Windows build configuration

DeJeune committed:
chore: update dependencies installation for macOS and Ubuntu in release workflow

DeJeune committed:
chore: add brew update step for macOS dependencies installation in release workflow

DeJeune committed:
Merge branch 'feat-knowlege-ocr' of https://github.com/CherryHQ/cherry-studio into feat/ocr

DeJeune committed:
chore: remove Windows dependencies from release workflow

DeJeune committed:
chore: update app-builder-lib patch and enhance installAppDependencies method

- Updated the resolution and checksum for the app-builder-lib patch in yarn.lock.
- Modified installAppDependencies method to accept excludeReBuildModules parameter.
- Adjusted platformPackager to pass excludeReBuildModules during dependency installation.
- Updated yarn.js to correctly reference excludeReBuildModules from options.

DeJeune committed:
chore: add Windows dependencies installation step in release workflow

- Introduced a new step to install necessary dependencies for Windows, including GTK and libjpeg-turbo.
- Updated the release workflow to ensure compatibility with Windows builds.

DeJeune committed:
chore: add canvas rebuild step for Windows in release workflow

- Introduced a new step to rebuild the canvas module for Windows to ensure compatibility during the build process.
- This step is conditionally executed based on the operating system in the workflow.

DeJeune committed:
chore: enhance canvas rebuild steps for Windows in release workflow

- Updated the canvas rebuild step to specify architecture for Windows x64 and added a new step for Windows ARM64.
- Ensures compatibility with both x64 and ARM64 architectures during the build process.

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

DeJeune committed:
refactor: remove canvas and pdfjs-dist dependencies and related image cropping logic

- Eliminated the canvas and pdfjs-dist dependencies from package.json and yarn.lock.
- Removed the image cropping functionality from BaseOcrProvider, simplifying the OCR process in MacSysOcrProvider.
- Updated the NSIS configuration in electron-builder.yml to reflect these changes.

DeJeune committed:
chore: add pdfjs-dist dependency to package.json and yarn.lock

- Added pdfjs-dist version 4.2.67 to both package.json and yarn.lock to support PDF rendering functionality.

DeJeune committed:
chore: enhance Windows build support in release workflow

- Added support for Windows 11 ARM architecture in the release workflow.
- Introduced a new step to install dependencies specific to Windows 11 ARM.
- Updated the canvas rebuild step to conditionally execute for Windows 11 ARM alongside existing Windows builds.

DeJeune committed:
chore: update dependencies and improve OCR functionality

- Added @cherrystudio/pdf-to-img-napi dependency to package.json and yarn.lock for enhanced PDF processing.
- Updated pdfjs-dist to version 4.10.38 for improved PDF rendering capabilities.
- Modified MacSysOcrProvider to utilize the new pdf-to-img-napi module for OCR processing of PDF files.
- Removed unnecessary Windows build steps from the release workflow to streamline the process.

DeJeune committed:
chore: update package.json to remove redundant canvas patches

- Removed duplicate canvas dependency patches from package.json.
- Retained the necessary patch for @langchain/core to ensure compatibility.

DeJeune committed:
chore: restore pdfjs-dist dependency in package.json

- Re-added pdfjs-dist version 4.10.38 to package.json to ensure PDF rendering functionality is available.

DeJeune committed:
chore: re-add pdfjs-dist dependency in package.json

- Restored pdfjs-dist version 4.10.38 to package.json to maintain PDF rendering functionality after its removal in a previous commit.

eeee0717 committed:
fix: use user key show unlimited

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
fix

eeee0717 committed:
Merge remote-tracking branch 'upstream/main' into feat-knowlege-ocr

eeee0717 committed:
fix

eeee0717 committed:
fix: search mcp cannot open

0xfullex commented:
这个是不是已经上线了……

eeee0717 committed:
chore: add mineru placeholder

eeee0717 committed:
Merge remote-tracking branch 'upstream/main' into feat-knowlege-ocr

eeee0717 committed:
fix

eeee0717 committed:
Merge remote-tracking branch 'upstream/main' into feat-knowlege-ocr

eeee0717 committed:
chore: change to 1.5.0-rc.1

eeee0717 committed:
fix: knowledge item cannot click

eeee0717 committed:
fix: tei error

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
fix: add macos-latest dependencies

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
fix: name and embedding model is required

eeee0717 committed:
feat: change knowledge style

alephpiece commented on src/renderer/src/pages/settings/ToolSettings/WebSearchSettings/WebSearchSettings/CompressionSettings/CutoffSettings.tsx:
这个文件好像是多余的

eeee0717 replied:
已移除

alephpiece commented on src/main/ocr/BaseOcrProvider.ts:
```diff
@@ -0,0 +1,122 @@
+
+  public async sendOcrProgress(sourceId: string, progress: number): Promise<void> {
+    const mainWindow = windowService.getMainWindow()
+    mainWindow?.webContents.send('file-preprocess-progress', {
```
这里发送了 `file-preprocess-progress` 却不是 `file-ocr-progress`，是有意的吗？

alephpiece commented on src/main/ocr/OcrProvider.ts:
```diff
@@ -0,0 +1,26 @@
+import { FileMetadata, PreprocessProvider as Provider } from '@types'
```
这里用了 `PreprocessProvider` 而不是 `OcrProvider`，虽然它们是一样的。

alephpiece replied:
总感觉 OCR 和 Preprocess 这个二分法不是特别好理解，Preprocess 这个名字也非常宽泛...
看起来它们都是 OCR，只不过服务来源不同，输出格式不同？

eeee0717 replied:
preprocess目前是针对文档预处理服务商，直接将pdf格式进行处理后导出markdown格式；ocr服务的设想是输入文档、图片等格式经过处理直接导出为文本，后续在聊天中会集成ocr服务。

YinsenHo replied:
@alephpiece 
preprocess 可以理解为一个工具集合的 pipline，OCR 只是其中之一的工具。preprocess 还包括了公式的提取，复杂排版，版面的空间结构，等等等，目标是把 pdf 等格式在尽量保留原有信息与关系的基础上，转成对 LLM 友好的 Markdown 格式。
OCR 的定义就非常精确且窄，就是把图像中的文字信息提取为文本。

eeee0717 committed:
fix: remove unnecessary file

eeee0717 committed:
fix: solve ocr ipc send and type error

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

alephpiece commented:
网络搜索页面宽度好像会溢出，现在多了一列了
![image](https://github.com/user-attachments/assets/1a3693d7-76fd-4e67-b375-09942d4652dd)

eeee0717 committed:
fix: ui problem

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
fix: remove opendal

eeee0717 commented on yarn.lock:
```diff
@@ -14196,6 +14419,36 @@ __metadata:
   languageName: node
   linkType: hard
 
+"opendal@npm:0.47.11":
```
ok

eeee0717 committed:
chore: adjust migrate

eeee0717 commented:
> migrate还需要调整吗？

ok了

eeee0717 committed:
chore

eeee0717 committed:
chore: change version

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
chore

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
Revert "Merge branch 'main' into feat-knowlege-ocr"

This reverts commit 5e09eceb9ed48616e79d8b53b6ac9460622dd14e.

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

alephpiece commented on src/renderer/src/pages/knowledge/components/QuotaTag.tsx:
```diff
@@ -0,0 +1,66 @@
+      return
+    }
+    checkQuota()
+  }, [_quota, base])
```
这有个warning

eeee0717 committed:
fix: remove patch

eeee0717 committed:
fix: quotaTag warning

eeee0717 committed:
Merge branch 'main' into feat-knowlege-ocr

eeee0717 committed:
fix: app-builder-lib

eeee0717 committed:
fix: electron vite

DeJeune commented on electron.vite.config.ts:
```diff
@@ -30,7 +30,7 @@ export default defineConfig({
     },
     optimizeDeps: {
-      noDiscovery: process.env.NODE_ENV === 'development'
+      disabled: 'dev'
```
可以撤回

eeee0717 committed:
revert: electron vite

DeJeune committed:
refactor: update GeminiAPIClient to streamline file handling

- Replaced the file retrieval and upload methods with direct API calls to improve efficiency.
- Removed unused methods related to file management, simplifying the codebase.
- Introduced createPartFromUri for better file data handling upon retrieval and upload.

eeee0717 commented on src/renderer/src/pages/knowledge/components/AddKnowledgePopup.tsx:
```diff
@@ -110,46 +136,48 @@ const PopupContainer: React.FC<Props> = ({ title, resolve }) => {
+          name: newBase.name,
           model: selectedEmbeddingModel,
           rerankModel: selectedRerankModel,
-          dimensions: autoDims ? undefined : values.dimensions,
```
应该是漏了，提个pr加一下吧

EurFelux commented on src/main/ipc.ts:
```diff
@@ -377,9 +378,10 @@ export function registerIpc(mainWindow: BrowserWindow, app: Electron.App) {
   ipcMain.handle(IpcChannel.File_Clear, fileManager.clear)
   ipcMain.handle(IpcChannel.File_Read, fileManager.readFile)
   ipcMain.handle(IpcChannel.File_Delete, fileManager.deleteFile)
+  ipcMain.handle('file:deleteDir', fileManager.deleteDir)
```
为啥不在IpcChannel里面加一个

eeee0717 replied:
之前写这个的时候还没有ipcchannel，合并的时候没有注意到
